src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultObjectValidator.cs

```diff
@@ -14,34 +15,36 @@ public class DefaultObjectValidator : IObjectModelValidator
     {
         private readonly IModelMetadataProvider _modelMetadataProvider;
         private readonly ValidatorCache _validatorCache;
+        private readonly IModelValidatorProvider _validatorProvider;
 
         /// <summary>
         /// Initializes a new instance of <see cref="DefaultObjectValidator"/>.
         /// </summary>
         /// <param name="modelMetadataProvider">The <see cref="IModelMetadataProvider"/>.</param>
-        /// <param name="validatorCache">The <see cref="ValidatorCache"/>.</param>
```

Comment:
Because it would never be used anywhere else, and it can just construct the cache. Putting the cache in DI just adds complexity